src/segment-loader.js

```diff
@@ -155,6 +155,9 @@ export default videojs.extend(videojs.EventTarget, {
 
   dispose() {
     this.abort_();
+    if (this.sourceUpdater_) {
+      this.sourceUpdater_.dispose();
```

Comment:
idk if this would be any better, but could shorten to one line with 
```javascript
if (this.sourceUpdater_ && this.sourceUpdater_.dispose()){}
```
Not sure if this is bad practice tho :p